### DISCUSSION TITLE
When I finish compiling, the tools are not generated in the folder，why?

### DISCUSSION THREAD

**Discussion by kinbiao** [azerothcore/azerothcore-wotlk#13719] [General]
When I finish compiling, the tools are not generated in the folder，why?

mapextractor.exe
mmaps_generator.exe
vmap4extractor.exe
vmap4assembler.exe

**Comment by virgo77** [comment]
Hi,
you need to specify the Tools buildings in your CMAKE command. You can follow this [wiki page](https://www.azerothcore.org/wiki/linux-core-installation) where you will have the DTOOLS_BUILD option.

In addition, you can just specify you only need the Map Tools, like this: `-DTOOLS_BUILD=maps-only` (thanks again to [Winfidonarleyan](https://github.com/azerothcore/azerothcore-wotlk/discussions/12077#discussioncomment-2976309) :))
So your CMAKE command will be like this: `cmake ../ -DCMAKE_INSTALL_PREFIX=$HOME/yourserverfolder/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DWITH_WARNINGS=1 -DTOOLS_BUILD=maps-only -DSCRIPTS=static -DMODULES=static`



See you!

V.

**Comment by kinbiao** [comment]
thank&nbsp; you&nbsp; very much.



---Original---
From: ***@***.***&gt;
Date: Sun, Nov 13, 2022 20:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [azerothcore/azerothcore-wotlk] When I finish compiling, the tools are not generated in the folder，why? (Discussion #13719)




 
Hi,
 you need to specify the Tools buildings in your CMAKE command. You can follow this wiki page where you will have the DTOOLS_BUILD option.
 
In addition, you can just specify you only need the Map Tools, like this: -DTOOLS_BUILD=maps-only (thanks again to Winfidonarleyan :))
 So your CMAKE command will be like this: cmake ../ -DCMAKE_INSTALL_PREFIX=$HOME/yourserverfolder/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DWITH_WARNINGS=1 -DTOOLS_BUILD=maps-only -DSCRIPTS=static -DMODULES=static
 
See you!
 
V.
